docs/src/single-validator-manager.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Single Validator Stake Pool Manager Program
+---
+
+A work-in-progress program for permissionless management of stake pools that are
+only delegated to one validator.
+
+## Operation
+
+The single-validator manager program defines a canonical stake pool for any
+validator on the network, which can be created by anyone.
+
+The program manages the "delegation strategy" of the pool, which is to maximize
+the amount of stake on one validator, while maintaining a small amount of SOL
+in the reserve for small stakers.
+
+### Create Pool
+
+The `CreateStakePool` instruction allows anyone to create the canonical stake
+pool for a validator, at the cost of ~0.0124 SOL for account rent-exemption of
+the stake pool, validator list, reserve, mint, and fee account.
+
+After the pool is created, the reserve requires at least stake rent-exemption plus
```

Comment:
wdyt about making the min reserve a (validator) configurable fraction of the pool size? so something like ...
```
MR = MIN(MD+RER, RF*S)

MR: min reserve
MD: min delegation
RER: rent-exempt reserve
RF: reserve fraction
S: supply
```

idea being that the larger the pool, the more likely the reserve will need liquidity over a given epoch. smaller and shrinking pools will collapse into the `MD+RER` case, so an awol validator won't lock anyone into the pool, regardless the `RF`

`RF` should likely be epoch-bound.

Reply:
I think you mean `MAX`, right?

And yeah, we could even apply some formula based on the size of the pool no matter what. I'm not sure what's best here.

Reply:
> I think you mean MAX, right?

yep. i'm no math man :sweat_smile:

Reply:
i think its better not introducing any configuration if possible, since our goal is one size fits all. and keeping the reserve target between the minimum stake and twice the minimum stake would be ideal from a capital efficiency perspective. the goal is for the majority of stakers to go to the pool, so we want it to be economically rational to do that

theres a lot of tricky ways to grief smallholders tho. like, say MD is 1sol. every epoch, someone deposits into the reserve to bring it to 2sol and initiates a 1sol activation. they then withdraw the remaining 1sol reserve, preventing smallholders from ever being able to withdraw. the cost of doing this is negligible even if we add a nominal withdrawal fee, though it gets more expensive as MD rises

actually they could avoid the deposit fee by, every epoch, initiating an activation if the reserve is over 2sol, depositing the equivalent of the remaining reserve balance as stake, and withdrawing the same amount as sol. so probably we need a withdrawal fee after all

then again the smallholder last resort is to sell on the market, so maybe we dont need to worry too much about griefers clogging the tubes? it affects no one who stakes above MD, inconveniences those who stake above MD at best, and if the fees accrue to the pool, maybe its just free rewards to stakers anyway?

Reply:
#3924 addresses one griefing attack at the stake pool level, so you can do multiple activations in an epoch, and allow us to bring the fee down to one epoch of rewards. Still not sure what to do about griefing small withdrawals without impacting pool performance...